src/loaders/OmeZarrLoader.ts

```diff
@@ -409,15 +398,18 @@ class OMEZarrLoader implements IVolumeLoader {
     }
 
     // Update volume `imageInfo` to reflect potentially new dimensions
-    const volsize = convertLoadSpecRegionToPixels(this.maxExtent, levelShape[xi], levelShape[yi], levelShape[zi]);
-    const [vx, vy, vz] = getExtentSize(volsize);
-    const offset = convertLoadSpecRegionToPixels(vol.loadSpec, vx, vy, vz);
+    const volRegion = convertSubregionToPixels(
+      this.maxExtent,
+      new Vector3(levelShape[xi], levelShape[yi], levelShape[zi])
+    );
+    const volSize = volRegion.getSize(new Vector3());
+    const offset = convertSubregionToPixels(vol.loadSpec.subregion, volSize);
     vol.imageInfo = {
       ...vol.imageInfo,
       atlasTileDims: new Vector2(nrows, ncols),
-      volumeSize: new Vector3(vx, vy, vz),
-      subregionSize: new Vector3(tw, th, tz),
-      subregionOffset: new Vector3(offset.minx, offset.miny, offset.minz),
+      volumeSize: volSize,
+      subregionSize: pxSize,
+      subregionOffset: offset.min,
```

Comment:
Nit: all 3 of the above vars are in pixels... would be good if the names all had px in them, like maybe volSizePx, subregionSizePx, offsetPx?

Reply:
As in the `ImageInfo` fields or the variables they're being set to?

Reply:
I was thinking `volSize`, `pxSize`, and `offset.min`